tests/zeromq.terminal/Program.cs

```diff
@@ -19,58 +20,60 @@ static Program()
         {
             _terminalActions = new Dictionary<string, Func<SocketConfiguration, Task>>
             {
-                ["pubsub"] = RunPubSub,
                 ["reqrep"] = RqRepScenarios.RunReqRep,
                 ["cancel"] = RqRepScenarios.CancellationTokenOnRunningTask,
                 ["contest"] = RqRepScenarios.Contest,
                 ["async-server"] = RqRepScenarios.AsyncServer,
-                ["bus-all"] = RqRepScenarios.UseBusInterface
+                ["bus-all"] = RqRepScenarios.UseBusInterface,
+
+                ["pubsub"] = PubSubScenarios.SimplePubSub,
+                ["subcancel"] = PubSubScenarios.PubSubWithCancellation,
             };
         }
 
         static async Task Main(string[] args)
         {
-            var version = args.FirstOrDefault();
-            var configuration = SocketConfiguration.InprocConfig($"this-inproc-sir-{Guid.NewGuid()}");
-            configuration.TimeOut = TimeSpan.FromSeconds(2);
-            
-            string key = "contest";
-            var actions = args.Where(x => _terminalActions.ContainsKey(x)).Select(x => _terminalActions[x]).ToList();
+            string key = args.First();
+            var actions = args.Where(x => _terminalActions.ContainsKey(x)).Select((x, index) => 
+            {
+                var configuration = BuildConfig(args, index);
+                return ExecuteScenario(x, configuration);
+            }
+            ).ToList();
             if (actions.Any())
             {
-                foreach (var action in actions)
+                foreach (var task in actions)
                 {
-                    await action(configuration);
+                    await task;
                 }
             }
             else
             {
-                await _terminalActions[key](configuration);
+                if (_terminalActions.ContainsKey(key))
+                    await ExecuteScenario(key, BuildConfig(args, 5));
```

Comment:
Yeah I had the same thought, but there is also the line with args.where(...) and there it reads horrible :D

I would leave it like it is, because it will always be a small dictionary with only 50 something entries or what not when all paterns are implemented, so the performance penalty for this is really minor.